.clang-format

```diff
@@ -0,0 +1,151 @@
+---
+Language:        Cpp
+# BasedOnStyle:  Google
```

Comment:
Are all of these options different from Google?  If not, does it work to just override the ones that have changed?

Reply:
no, just a list of all the options from:
$clang-format-8 -style=google -dump-config > .clang-format
this just dumps all the configurations available, already tuned with the google style.
It's useful to have all the configs to know exactly what are the options available if we need to change anything.
I can add in a comment on which ones were changed from the -dump-config default.

Reply:
Ok that would be helpful

Reply:
I did that in the last commit " highlight the configs that were changed from the original -dump-config in the google style"
thanks!